### PR TITLE
Convert to GitHub Actions

### PR DESCRIPTION
**Pull request by gagelarsen**
- Add GitHub actions
- Remove Travis CI
image

### CODE REVIEW

File: .github/workflows/tethys-release.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          . ~/miniconda/etc/profile.d/conda.sh;
+          conda activate tethys;
+          conda build -c tethysplatform -c conda-forge ./tethys/conda.recipe
+      # Upload Conda No Pull Request With Tag
+      - name: Upload Conda With Tag
+        run: |
+          cd ..
+          . ~/miniconda/etc/profile.d/conda.sh;
+          conda activate tethys;
+          anaconda -t "${{ secrets.CONDA_UPLOAD_TOKEN }}" upload -u aquaveo $CONDA_BLD_PATH/noarch/tethys-platform*.tar.bz2 --force;
```
Comment (swainn) [inline .github/workflows/tethys-release.yml:98]: The user `aquaveo` is hard coded. Swap it out for an environment variable: CONDA_UPLOAD_USER

File: .github/workflows/tethys-release.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      # Checkout the source
+      - name: Checkout Source
+        uses: actions/checkout@v2
+      # Get Tag Name
+      - name: Get Tag
+        id: gitTag
+        uses: little-core-labs/get-git-tag@v3.0.2
+      # Build the docker for tag
+      - name: Build With Tag
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
```
Comment (swainn) [inline .github/workflows/tethys-release.yml:32]: Needed?

File: .github/workflows/tethys-release.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        run: |
+               docker build -t ${{ secrets.DOCKER_UPLOAD_URL }}:latest -t ${{ secrets.DOCKER_UPLOAD_URL }}:${{ steps.gitTag.outputs.tag }} .;
+      # Upload docker
+      - name: Upload Docker With Tag
+        if: ${{ github.event_name != 'pull_request' }} && ${{ startsWith(github.ref, 'refs/tags/') }} == "True"
```
Comment (swainn) [inline .github/workflows/tethys-release.yml:37]: Needed?

File: .github/workflows/tethys-release.yml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  push:
+    tags:
+      - '*'
+
+env:
+  CONDA_BUILD_PIN_LEVEL: minor
+
+jobs:
+  tethys-docker:
+    name: Tethys Docker (${{ matrix.platform }})
```
Comment (swainn) [inline .github/workflows/tethys-release.yml:15]: `tethys-docker` => `docker-build`
"Tethys Docker" => "Docker Build"

File: .github/workflows/tethys-release.yml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      # Upload docker
+      - name: Upload Docker With Tag
+        if: ${{ github.event_name != 'pull_request' }} && ${{ startsWith(github.ref, 'refs/tags/') }} == "True"
+        run: |
+          echo "Pushing to docker registry";
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin;
+          docker push ${{ secrets.DOCKER_UPLOAD_URL }}:${{ steps.gitTag.outputs.tag }};
+
+  tethys-conda:
+    name: Tethys Conda (${{ matrix.platform }})
```
Comment (swainn) [inline .github/workflows/tethys-release.yml:44]: `tethys-conda` => `conda-build`
"Tethys Conda" => "Conda Build"

File: .github/workflows/tethys.yml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    branches:
+      - '*'
+
+env:
+  CONDA_BUILD_PIN_LEVEL: minor
+
+jobs:
+
+  tethys-tests:
+    name: Tethys Test (${{ matrix.platform }})
```
Comment (swainn) [inline .github/workflows/tethys.yml:19]: `tethys-tests` => `tests`
"Tethys Test" => "Tests"

File: .github/workflows/tethys.yml [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+      # Generate Coverage Report
+      - name: Genearte Coverage Report
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+            . ~/miniconda/etc/profile.d/conda.sh
+            conda activate tethys
+            coveralls
+
+  tethys-docker:
+    name: Tethys Docker (${{ matrix.platform }})
```
Comment (swainn) [inline .github/workflows/tethys.yml:59]: Same as above

File: .github/workflows/tethys.yml [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin;
+          docker push ${{ secrets.DOCKER_UPLOAD_URL }}:master;
+      # No Upload if Pull Request
+      - name: No Upload
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          echo "Uploading is skipped for pull requests."
+
+  tethys-conda:
+    name: Tethys Conda (${{ matrix.platform }})
```
Comment (swainn) [inline .github/workflows/tethys.yml:94]: Same as above

File: .github/workflows/tethys.yml [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+          conda build -c tethysplatform -c conda-forge ./tethys/conda.recipe
+      # Upload Conda No Pull Request No Tag
+      - name: Upload Conda No Tag
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          cd ..
+          . ~/miniconda/etc/profile.d/conda.sh;
+          conda activate tethys;
+          ls ~/conda-bld/noarch
+          anaconda -t "${{ secrets.CONDA_UPLOAD_TOKEN }}" upload -u aquaveo -l dev $CONDA_BLD_PATH/noarch/tethys-platform*.tar.bz2 --force;
```
Comment (swainn) [inline .github/workflows/tethys.yml:156]: The user aquaveo is hard coded. Swap it out for an environment variable: CONDA_UPLOAD_USER

**Review by swainn — CHANGES_REQUESTED** [review]
Added a few comments

File: .github/workflows/tethys.yml (hunk 9 above)
Comment (gagelarsen) [reply .github/workflows/tethys.yml:156]: Done

File: .github/workflows/tethys.yml (hunk 8 above)
Comment (gagelarsen) [reply .github/workflows/tethys.yml]: fixed

File: .github/workflows/tethys.yml (hunk 7 above)
Comment (gagelarsen) [reply .github/workflows/tethys.yml]: fixed

File: .github/workflows/tethys.yml (hunk 6 above)
Comment (gagelarsen) [reply .github/workflows/tethys.yml]: fixed

File: .github/workflows/tethys-release.yml (hunk 5 above)
Comment (gagelarsen) [reply .github/workflows/tethys-release.yml]: fixed

File: .github/workflows/tethys-release.yml (hunk 4 above)
Comment (gagelarsen) [reply .github/workflows/tethys-release.yml]: fixed

File: .github/workflows/tethys-release.yml (hunk 3 above)
Comment (gagelarsen) [reply .github/workflows/tethys-release.yml]: removed

File: .github/workflows/tethys-release.yml (hunk 2 above)
Comment (gagelarsen) [reply .github/workflows/tethys-release.yml]: Removed

File: .github/workflows/tethys-release.yml (hunk 1 above)
Comment (gagelarsen) [reply .github/workflows/tethys-release.yml]: done

**Review by swainn — APPROVED** [review]

**Pull request merged**